source/extensions/omni.isaac.lab/omni/isaac/lab/scene/interactive_scene.py

```diff
@@ -355,6 +387,37 @@ def reset(self, env_ids: Sequence[int] | None = None):
         for sensor in self._sensors.values():
             sensor.reset(env_ids)
 
+    def reset_to(
+        self, state: dict[str, dict[str, torch.Tensor]], env_ids: Sequence[int] | None = None, is_relative: bool = False
+    ):
+        """Resets the scene entities to the given state.
+
+        Args:
+            state: The state to reset the scene entities to.
+            env_ids: The indices of the environments to reset.
+                Defaults to None (all instances).
+            is_relative: If set to True, the state is considered relative to the environment origins.
+        """
+        if env_ids is None:
+            env_ids = slice(None)
+        # -- assets
+        for asset_name, articulation in self._articulations.items():
+            joint_position = torch.Tensor(state["articulation"][asset_name])
+            articulation.write_joint_state_to_sim(joint_position, torch.zeros_like(joint_position), env_ids=env_ids)
+            articulation.set_joint_position_target(joint_position, env_ids=env_ids)
+            articulation.set_joint_velocity_target(torch.zeros_like(joint_position), env_ids=env_ids)
```

Comment:
as with the comments in scene.get_state we will need to set velocity and effort here too.,

Reply:
Added velocity in the getter/reset_to state.
The reset_to method is expected to be invoked to one time in the beginning before replaying a pre-recorded episode to make sure the asset's initial states match the recorded states. Would it still be necessary to set effort here in this case?

Reply:
It may not be 100% necessary but if the effort is initialized to something like 'zeros' this may cause a divergent behavior. It may be really minor.

Reply:
If this is only used for 'replay' i.e. setting the position and velocity at each timestep from file then this will likely not be necessary.